.circleci/config.yml

```diff
@@ -1,80 +1,151 @@
+          project: "Presentation.xcodeproj"
+          scheme: "Presentation"
+
+  test_example_project:
+    parameters:
+      path: # We need to pass a path here because the example projects are in a sub folder of the project
+        type: string
     steps:
       - checkout
+      - pod-install:
+          path: <<parameters.path>>
+      - test_workspace_and_store_results:
+          workspace: "Example.xcworkspace"
+          scheme: "Example"
+          path: <<parameters.path>>
+
+  # We introduced two separate commands for projects and workspaces because we didnt find a generic and non-confusing way to introduce 
+  # a condition to only pass either the project or the workspace environment argument to the fastlane scan
+  test_project_and_store_results:
+    description: "Builds and tests a project and then stores the results of the tests as artifacts and test results report"
+    parameters:
+      project:
+        type: string
+      scheme:
+        type: string
+    steps:
       - run:
-          name: test iOS
-          command: |
-            set -o pipefail
-            xcodebuild -version
-            xcodebuild -showsdks
-            swift -version
-            IOS_SDK="iphonesimulator12.2" \
-                IOS_DESTINATION_PHONE="OS=12.2,name=iPhone Xs" \
-                sh build.sh test-iOS
+          command: fastlane scan
+          environment:
+              SCAN_PROJECT: <<parameters.project>>
+              SCAN_SCHEME: <<parameters.scheme>>
+              SCAN_DEVICE: *test_device
+              SCAN_CLEAN: *clean_before_build
+      - store_artifacts: # This will by default store an html and junit file as artifacts (See "Artifacts" tab in CircleCI report)
+          path: *test_output_folder # test_output is the default temporary folder for fastlane scan output
+          destination: *test_output_folder # This will create a sub structure in the artifacts section in CircleCI
+      - store_test_results: # This will store the test results so you can then see them in the "Test Summary" tab in CircleCI report
+          path: *test_output_folder
 
+  test_workspace_and_store_results:
+    description: "Builds and tests a workspace and then stores the results of the tests as artifacts and test results report"
+    parameters:
+      workspace:
+        type: string
+      scheme:
+        type: string
+      path:
+        type: string
+    steps:
+      - run:
+          command:  |
+            cd <<parameters.path>>
+            fastlane scan
+          environment:
+              SCAN_WORKSPACE: <<parameters.workspace>>
+              SCAN_SCHEME: <<parameters.scheme>>
+              SCAN_DEVICE: *test_device
+              SCAN_CLEAN: *clean_before_build
+      - store_artifacts: # This will by default store an html and junit file as artifacts (See "Artifacts" tab in CircleCI report)
+          path: *test_output_folder # test_output is the default temporary folder for fastlane scan output
```

Comment:
Seems like the storage of artefacts and test results is failing for the example jobs. Would be nice to get that in as well.